File: dotnet_projects/OcrLiteOnnxCs/OcrLiteOnnxForm/FormOcrLite.cs
using Emgu.CV;
using Emgu.CV.CvEnum;
using OcrLiteLib;
using System;
using System.IO;
using System.Windows.Forms;

namespace OcrLiteOnnxCs
{
    public partial class FormOcrLite : Form
    {
        private OcrLite ocrEngin;

        public FormOcrLite()
        {
            InitializeComponent();
        }

        private void FormOcrLite_Load(object sender, EventArgs e)
        {
            string appPath = AppDomain.CurrentDomain.BaseDirectory;
            string appDir = Directory.GetParent(appPath).FullName;
            string modelsDir = appPath + "models";
            modelsTextBox.Text = modelsDir;
            string detPath = modelsDir + "\\" + detNameTextBox.Text;
            string clsPath = modelsDir + "\\" + clsNameTextBox.Text;
            string recPath = modelsDir + "\\" + recNameTextBox.Text;
            string keysPath = modelsDir + "\\" + keysNameTextBox.Text;
            bool isDetExists = File.Exists(detPath);
            if (!isDetExists)
            {
                MessageBox.Show("模型文件不存在:" + detPath);
            }
            bool isClsExists = File.Exists(clsPath);
            if (!isClsExists)
            {
                MessageBox.Show("模型文件不存在:" + clsPath);
            }
            bool isRecExists = File.Exists(recPath);
            if (!isRecExists)
            {
                MessageBox.Show("模型文件不存在:" + recPath);
            }
            bool isKeysExists = File.Exists(keysPath);
            if (!isKeysExists)
            {
                MessageBox.Show("Keys文件不存在:" + keysPath);
            }
            if(isDetExists && isClsExists && isRecExists && isKeysExists)
            {
                ocrEngin = new OcrLite();
                ocrEngin.InitModels(detPath,clsPath,recPath,keysPath, (int)numThreadNumeric.Value);
            }
            else
            {
                MessageBox.Show("初始化失败，请确认模型文件夹和文件后，重新初始化！");
            }
        }

        private void initBtn_Click(object sender, EventArgs e)
        {
            string modelsDir = modelsTextBox.Text;
            string detPath = modelsDir + "\\" + detNameTextBox.Text;
            string clsPath = modelsDir + "\\" + clsNameTextBox.Text;
            string recPath = modelsDir + "\\" + recNameTextBox.Text;
            string keysPath = modelsDir + "\\" + keysNameTextBox.Text;
            bool isDetExists = File.Exists(detPath);
            if (!isDetExists)
            {
                MessageBox.Show("模型文件不存在:" + detPath);
            }
            bool isClsExists = File.Exists(clsPath);
            if (!isClsExists)
            {
                MessageBox.Show("模型文件不存在:" + clsPath);
            }
            bool isRecExists = File.Exists(recPath);
            if (!isRecExists)
            {
                MessageBox.Show("模型文件不存在:" + recPath);
            }
            bool isKeysExists = File.Exists(recPath);
            if (!isKeysExists)
            {
                MessageBox.Show("Keys文件不存在:" + keysPath);
            }
            if (isDetExists && isClsExists && isRecExists && isKeysExists)
            {
                ocrEngin = new OcrLite();
                ocrEngin.InitModels(detPath, clsPath, recPath, keysPath, (int)numThreadNumeric.Value);
            }
            else
            {
                MessageBox.Show("初始化失败，请确认模型文件夹和文件后，重新初始化！");
            }
        }

        private void openBtn_Click(object sender, EventArgs e)
        {
            using (var dlg = new OpenFileDialog())
            {
                dlg.Multiselect = false;
                dlg.Filter = "(*.JPG,*.PNG,*.JPEG,*.BMP,*.GIF)|*.JPG;*.PNG;*.JPEG;*.BMP;*.GIF|All files(*.*)|*.*";
                if (dlg.ShowDialog() == DialogResult.OK && !string.IsNullOrEmpty(dlg.FileName))
                {
                    pathTextBox.Text = dlg.FileName;
                    Mat src = CvInvoke.Imread(dlg.FileName, ImreadModes.Color);
                    pictureBox.Image = src.ToBitmap();
                }
            }
        }

        private void modelsBtn_Click(object sender, EventArgs e)
        {
            using (var dlg = new FolderBrowserDialog())
            {
                dlg.SelectedPath = Environment.CurrentDirectory + "\\models";
                if (dlg.ShowDialog() == DialogResult.OK && !string.IsNullOrEmpty(dlg.SelectedPath))
                {
                    modelsTextBox.Text = dlg.SelectedPath;
                }
            }
        }

        private void detectBtn_Click(object sender, EventArgs e)
        {
            if(ocrEngin == null)
            {
                MessageBox.Show("未初始化，无法执行!");
                return;
            }
            string targetImg = pathTextBox.Text;
            if (!File.Exists(targetImg))
            {
                MessageBox.Show("目标图片不存在，请用Open按钮打开");
                return;
            }
            int padding = (int)paddingNumeric.Value;
            int imgResize = (int)imgResizeNumeric.Value;
            float boxScoreThresh = (float)boxScoreThreshNumeric.Value;
            float boxThresh = (float)boxThreshNumeric.Value;
            float unClipRatio = (float)unClipRatioNumeric.Value;
            bool doAngle = doAngleCheckBox.Checked;
            bool mostAngle = mostAngleCheckBox.Checked;
            OcrResult ocrResult = ocrEngin.Detect(pathTextBox.Text, padding, imgResize, boxScoreThresh, boxThresh, unClipRatio, doAngle, mostAngle);
            ocrResultTextBox.Text = ocrResult.ToString();
            strRestTextBox.Text = ocrResult.StrRes;
            pictureBox.Image = ocrResult.BoxImg.ToBitmap();
        }

        private void partImgCheckBox_CheckedChanged(object sender, EventArgs e)
        {
            ocrEngin.isPartImg = partImgCheckBox.Checked;
        }

        private void debugCheckBox_CheckedChanged(object sender, EventArgs e)
        {
            ocrEngin.isDebugImg = debugCheckBox.Checked;
        }
    }
}

File: dotnet_projects/OcrLiteOnnxVB/OcrLiteOnnxForm/FormOcrLite.vb
Imports System.IO
Imports Emgu.CV
Imports Emgu.CV.CvEnum
Imports OcrLiteLib.OcrLiteLib

Public Class FormOcrLite
    Private ocrEngin As OcrLite
    Private Sub FormOcrLite_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Dim appPath As String = AppDomain.CurrentDomain.BaseDirectory
        Dim appDir As String = Directory.GetParent(appPath).FullName
        Dim modelsDir As String = appPath & Convert.ToString("models")
        modelsTextBox.Text = modelsDir
        ocrEngin = New OcrLite(modelsDir, CInt(numThreadNumeric.Value))
    End Sub

    Private Sub initBtn_Click(sender As Object, e As EventArgs) Handles initBtn.Click
        ocrEngin = New OcrLite(modelsTextBox.Text, CInt(numThreadNumeric.Value))
    End Sub

    Private Sub openBtn_Click(sender As Object, e As EventArgs) Handles openBtn.Click
        Using dlg = New OpenFileDialog()
            dlg.Multiselect = False
            dlg.Filter = "(*.JPG,*.PNG,*.JPEG,*.BMP,*.GIF)|*.JPG;*.PNG;*.JPEG;*.BMP;*.GIF|All files(*.*)|*.*"
            If dlg.ShowDialog() = DialogResult.OK AndAlso Not String.IsNullOrEmpty(dlg.FileName) Then
                pathTextBox.Text = dlg.FileName
                Dim src As Mat = CvInvoke.Imread(dlg.FileName, ImreadModes.Color)
                pictureBox.Image = src.ToBitmap()
            End If
        End Using
    End Sub

    Private Sub modelsBtn_Click(sender As Object, e As EventArgs) Handles modelsBtn.Click
        Using dlg = New FolderBrowserDialog()
            dlg.SelectedPath = Environment.CurrentDirectory + "\models"
            If dlg.ShowDialog() = DialogResult.OK AndAlso Not String.IsNullOrEmpty(dlg.SelectedPath) Then
                modelsTextBox.Text = dlg.SelectedPath
            End If
        End Using
    End Sub

    Private Sub detectBtn_Click(sender As Object, e As EventArgs) Handles detectBtn.Click
        Dim padding As Integer = CInt(paddingNumeric.Value)
        Dim imgResize As Integer = CInt(imgResizeNumeric.Value)
        Dim boxScoreThresh As Single = CSng(boxScoreThreshNumeric.Value)
        Dim boxThresh As Single = CSng(boxThreshNumeric.Value)
        Dim minArea As Single = CSng(minAreaNumeric.Value)
        Dim unClipRatio As Single = CSng(unClipRatioNumeric.Value)
        Dim doAngle As Boolean = doAngleCheckBox.Checked
        Dim mostAngle As Boolean = mostAngleCheckBox.Checked
        Dim ocrResult As OcrResult = ocrEngin.Detect(pathTextBox.Text, padding, imgResize, boxScoreThresh, boxThresh, minArea,
                unClipRatio, doAngle, mostAngle)
        ocrResultTextBox.Text = ocrResult.ToString()
        strRestTextBox.Text = ocrResult.StrRes
        pictureBox.Image = ocrResult.BoxImg.ToBitmap()
    End Sub

    Private Sub partImgCheckBox_CheckedChanged(sender As Object, e As EventArgs) Handles partImgCheckBox.CheckedChanged
        ocrEngin.isPartImg = partImgCheckBox.Checked
    End Sub

    Private Sub debugCheckBox_CheckedChanged(sender As Object, e As EventArgs) Handles debugCheckBox.CheckedChanged
        ocrEngin.isDebugImg = debugCheckBox.Checked
    End Sub
End Class
